src/jep/convert_p2j.c

```diff
@@ -52,6 +55,57 @@ static jmethodID unmodifiableList      = 0;
     static jstring UTF8 = NULL;
 #endif
 
+char isFunctionalInterfaceType(JNIEnv *env, jclass type) {
+    if (!JNI_METHOD(classGetMethods, env, JCLASS_TYPE, "getMethods", "()[Ljava/lang/reflect/Method;")) {
+        process_java_exception(env);
+        return 0;
+    }
+    if (!JNI_METHOD(classIsInterfaceMethod, env, JCLASS_TYPE, "isInterface", "()Z")) {
+        process_java_exception(env);
+        return 0;
+    }
+    if (!JNI_METHOD(methodGetModifiers, env, JMETHOD_TYPE, "getModifiers", "()I")) {
+        process_java_exception(env);
+        return 0;
+    }
+    if ((*env)->PushLocalFrame(env, JLOCAL_REFS) != 0) {
+        process_java_exception(env);
+        return 0;
+    }
+    if (!(*env)->CallBooleanMethod(env, type, classIsInterfaceMethod)) {
+        return 0; // It's not an interface, so it can't be functional
+    }
+    jobjectArray methods = (jobjectArray) (*env)->CallObjectMethod(env, type, classGetMethods);
```

Comment:
You have to declare all the variables at the top of the block.  It's annoying but required for some compilers.